### ISSUE TITLE
Unable to run sunpy.net.helio.HECClient().get_table_names(...) in Py3.x

### ISSUE THREAD

**Issue by m-charlton** [sunpy/sunpy#2043]
There seems to be a problem with using hec.py under Python 3.5. I suspect the 
root cause is that in P2.x `str` & `bytes` are the same type, in Py3.x they are
different.

#### How to reproduce

In Py3.5 environment, execute the following code:

```
>>> from sunpy.net.helio import hec
>>> hc = hec.HECClient()
>>> hc.get_table_names()

...
  markup_type=markup_type))
a bytes-like object is required, not 'str'
Traceback (most recent call last):
  File "/.../github/sunpy/env35/lib/python3.5/site-packages/suds/plugin.py", line 255, in __call__
    method(ctx)
  File "/.../github/sunpy/sunpy/net/helio/hec.py", line 129, in received
    self.last_payload = six.u(suds_unwrapper(context.reply))
  File "/.../github/sunpy/sunpy/net/helio/hec.py", line 77, in suds_unwrapper
    pos_1 = wrapped_data.find(CATCH_1)
TypeError: a bytes-like object is required, not 'str'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../github/sunpy/sunpy/net/helio/hec.py", line 235, in get_table_names
    tables = votable_handler(self.votable_interceptor.last_payload)
  File "/.../github/sunpy/sunpy/net/helio/hec.py", line 114, in votable_handler
    fake_file.write(xml_table)
TypeError: string argument expected, got 'NoneType'
```

#### On first inspection:

- `suds_unwrapper(...)` is expecting a String, but receives bytes 
- The `astrpoy.parse_single_table(...)` function called in `votable_handler(...)` is expecting `bytes` and not `str`
  
Working on a fix and tests.

**Comment by swapsha96** [comment]
I saw the source code(https://github.com/sunpy/sunpy/blob/master/sunpy/net/helio/hec.py) and found out that this error is because  StringIO is used in the file. Instead of this, BytesIO must be used. Previously, I used it to extract data from FITS file and display image using matplotlib in which I required 'io' and faced similar error. Can I start working on this and later, create a PR?

**Comment by swapsha96** [comment]
@m-charlton 

- decode("utf-8") can be used to convert bytes to String
- I checked the type of xml_table and it gives:
`<class 'NoneType'>`
Can you tell me what is the type supposed to be? Additionally BytesIO() must be used to extend compatibility with Python 3.x.

**Comment by nabobalis** [comment]
Closed by #2130
